task 1/main.cpp

```diff
@@ -0,0 +1,232 @@
+#include <opencv2/opencv.hpp>
+#include <iostream>
+#include <math.h>
+
+void CallBack_Demo(int, void*)
+{
+	return;
+}
+
+
+void get_lights_mask(cv::InputArray img, cv::OutputArray mask)
+{
+	cv::Mat imgHSV, imgBlur, maskBlur;;
+	int h_min = 0, h_max = 255, s_min = 0, s_max = 159, v_min = 231, v_max = 255;
+
+	// 转HSV格式
+	cv::cvtColor(img, imgHSV, cv::COLOR_BGR2HSV);
+	// 高斯模糊
+	cv::GaussianBlur(imgHSV, imgBlur, cv::Size(9, 9), 0);
+	// 以指定范围的阈值进行二值化
+	cv::inRange(imgBlur, cv::Scalar(h_min, s_min, v_min), cv::Scalar(h_max, s_max, v_max), maskBlur);
+	// 优化处理
+	// 模糊
+	cv::GaussianBlur(maskBlur, maskBlur, cv::Size(7, 7), 0);
+	// 闭操作填充黑点
+	cv::morphologyEx(maskBlur, mask, cv::MORPH_CLOSE, cv::Mat(9, 9, CV_8U));
+
+
+}
+
+void get_lights_contours(cv::InputOutputArray mask, std::vector<std::vector<cv::Point>> &contours)
+{
+	// 提取轮廓
+	cv::findContours(mask, contours, cv::RETR_EXTERNAL, cv::CHAIN_APPROX_NONE);
+}
+
+void get_lights_rects(std::vector<std::vector<cv::Point>> contours, std::vector<cv::RotatedRect> &rects)
```

Comment:
you can use some more informative names for your variables, such as `light_rects` instead of `rects`.